src/components/GlobalPreferences/Notifications/SubscriptionsForm.js

```diff
@@ -0,0 +1,212 @@
+    app => `${app.name} ${app.identifier ? `(${app.identifier})` : ''}`
+  )
+  const selectedApp =
+    selectedAppIdx === -1 ? null : subscribableApps[selectedAppIdx]
+
+  let eventNames = ['']
+
+  if (selectedApp) {
+    // Once an app is selected, get corresponding events
+    eventNames = subscribableEvents[selectedAppIdx]
+    if (eventNames.length === 0) {
+      // if subscribed to all events reset the app
+      setSelectedAppIdx(-1)
+    }
+  }
+
+  const handleAppChange = index => {
+    setSelectedAppIdx(index)
+    setSelectedEventIdx(-1)
+  }
+  const handleEventChange = index => {
+    setSelectedEventIdx(index)
+  }
+  const handleSubscribe = async e => {
+    setIsSubmitting(true)
+    const { abi, appName, proxyAddress } = selectedApp
+    const eventName = eventNames[selectedEventIdx]
+
+    const abiSubset = [
+      abi.find(({ name, type }) => type === 'event' && name === eventName),
+    ]
+
+    try {
+      const payload = {
+        abi: abiSubset,
+        appName,
+        appContractAddress: proxyAddress,
+        ensName: dao,
+        eventName: eventNames[selectedEventIdx],
+        network: getEthNetworkType(),
+        token,
+      }
+      await createSubscription(payload)
+      setSelectedEventIdx(-1)
+      setSelectedAppIdx(-1) // Reset app as it may be unavailable if subscribed to all that app's events
+      onCreate()
+    } catch (e) {
+      onApiError(e.message)
+    }
+    setIsSubmitting(false)
+  }
+  const isSubscribeDisabled =
+    selectedAppIdx === -1 || selectedEventIdx === -1 || isSubmitting
+
+  if (isFetchingSubscriptions) {
+    return (
+      <Box heading="Create Subscriptions">
+        <LoadingRing />
+      </Box>
+    )
+  }
+
+  if (subscribableApps.length === 0 && !isFetchingSubscriptions) {
+    return (
+      <Box heading="Create Subscriptions">You are subscribed to all events</Box>
+    )
+  }
+
+  return (
+    <Box heading="Create Subscriptions">
+      <div
+        css={`
+          margin-bottom: ${2 * GU}px;
+        `}
+      >
+        <Label
+          css={`
+            color: ${theme.surfaceContentSecondary};
+          `}
+        >
+          App
+        </Label>
+        <DropDown
+          wide
+          placeholder="Select an App"
+          items={appNames}
+          selected={selectedAppIdx}
+          onChange={handleAppChange}
+        />
+      </div>
+      <div>
+        <Label
+          css={`
+            color: ${theme.surfaceContentSecondary};
+          `}
+        >
+          Event
+        </Label>
+        <DropDown
+          disabled={selectedAppIdx === -1}
+          placeholder="Select an Event"
+          width="100%"
+          items={eventNames}
+          selected={selectedEventIdx}
+          onChange={handleEventChange}
+        />
+      </div>
+      <div
+        css={`
+          margin-top: ${4 * GU}px;
+        `}
+      >
+        <Button disabled={isSubscribeDisabled} onClick={handleSubscribe}>
```

Comment:
Does it make sense to persist the app selected when subscribing so that the user does not have to re-select the app?
https://www.loom.com/share/e36b1dee83dc4f779a014de9efa0bc56

Reply:
Initially, I had it that way. The reason I no longer do it is because we re-render the list of apps after creating a subscription and if an app is removed from the list, it will select a different app with the same index (because the apps array is different and the index hasn't changed).

Reply:
Ahhhh, I saw this when unsubscribing, gotcha.

Reply:
It's possible but tricky. The list of available apps can change upon subscription, hence we reset the selection so that it doesn't jump in case the selected app is removed.

Reply:
Maybe persist the selection and only remove it when the last event has been subscribed to?

Reply:
Fixed in https://github.com/aragon/aragon/pull/904/commits/c640eeb4f12d8e7a4e1b88e3603dbed16d8e0d18